pkg/actors/actors.go

```diff
@@ -296,12 +296,24 @@ func (a *actorsRuntime) callRemoteActorWithRetry(
 	return nil, errors.Errorf("failed to invoke target %s after %v retries", targetAddress, numRetries)
 }
 
+func (a *actorsRuntime) getOrCreateActor(actorType, actorID string) *actor {
+	key := a.constructCompositeKey(actorType, actorID)
+
+	// This avoids allocating multiple actor allocations by calling newActor
+	// whenever actor is invoked. When storing actor key first, there is a chance to
+	// call newActor, but this is trivial.
+	val, ok := a.actorsTable.Load(key)
```

Comment:
Do we worry about a race condition here? Previously LoadOrStore is concurrency safe call. Our new check-and-set block is not atomic.

Reply:
I see, there is not much harm from race here explained in the other comments.

Reply:
The point is here. I am worry about race condition of actorsTable since we use sync.map. 

What I am pointing it out here actor object allocation, which is the unnecessary mem alloc. The master code alloc actor object whenever actor is invoked. this is what I avoid here.

https://github.com/dapr/dapr/blob/b8e8eecb5fbb042f1e1246396e021abe55204829/pkg/actors/actor.go#L28-L37